### ISSUE TITLE
irony for json cdb don't support "arguments" key in compile_commands.json

### ISSUE THREAD

**Issue by ANDRON94** [Sarcasm/irony-mode#403]
Hi, I have the next json compilation database file: 

> [ 
    {
        "arguments": [
            "c++", 
            "-c", 
            "-DHAVE_CONFIG_H", 
            "-I.", 
            "-I../../src", 
            "-I..", 
            "-pthread", 
            "-I/usr/include", 
            "-pthread",
            "OTHER STUFF"
        ], 
        "directory": "/home/XXXXXX", 
        "file": "./mgmt/XXXXX.cpp"
    }
]

And it  didn't properly loaded into Emacs. I've got next error message:

> Irony CDB: error in compilation database: (irony-server-error cannot-load-database "failed to load compilation database from directory" "/home/XXXXX").

I inspect an Irony code a little bit(irony-cdb-json--compile-command-options) and if I don't miss something the current implementation only support "command" key not "arguments". But  from [JSON Compilation Database specification](http://clang.llvm.org/docs/JSONCompilationDatabase.html):

> *arguments*: The compile command executed as list of strings. Either arguments or command is required.

**Comment by Sarcasm** [comment]
Indeed, this feature has been added somewhat recently to the compilation database specification.
It could work with the libclang backend if your clang version is up-to-date.
The JSON parsing done in Elisp needs to be updated, it should not be too difficult to contribute ;)

**Comment by ANDRON94** [comment]
Yeah, I'll create a pull request asap.
